src/data/list/basic.lean

```diff
@@ -132,6 +132,11 @@ end
   f a ∈ map f l ↔ a ∈ l :=
 ⟨λ m, let ⟨a', m', e⟩ := exists_of_mem_map m in H e ▸ m', mem_map_of_mem _⟩
 
+@[simp] lemma map_nil (f : α → β) : list.map f [] = [] := rfl
```

Comment:
`map` is already labelled as `simp`. We don't need that lemma. You can use this as a proof for the next lemma:

```lean
@[simp] lemma map_eq_nil {f : α → β} {l : list α} : list.map f l = [] ↔ l = [] :=
⟨by cases l; simp, λ h, h.symm ▸ rfl⟩
```

Reply:
Also, can you use `squeeze_simp`?